sdk/eventhub/Azure.Messaging.EventHubs/perf/Azure.Messaging.EventHubs.Perf/Azure.Messaging.EventHubs.Perf/Scenarios/Standalone/ConsumeEvents.cs

```diff
@@ -0,0 +1,172 @@
+            {
+                clients[i] = new EventHubConsumerClient(connectionString, _eventHubName);
+            }
+
+            try
+            {
+                var client = clients.First();
+                var partitionIds = (await client.GetPartitionIdsAsync()).Take(numPartitions);
+                var partitions = await Task.WhenAll(partitionIds.Select(id => client.GetPartitionPropertiesAsync(id)));
+
+                var totalCount = (long)0;
+                foreach (var partition in partitions)
+                {
+                    var begin = partition.BeginningSequenceNumber;
+                    var end = partition.LastEnqueuedSequenceNumber;
+                    var count = end - begin + 1;
+                    totalCount += count;
+
+                    if (verbose)
+                    {
+                        Console.WriteLine($"Partition: {partition.Id}, Begin: {begin}, End: {end}, Count: {count}");
+                    }
+                }
+                if (verbose)
+                {
+                    Console.WriteLine($"Total Count: {totalCount}");
+                }
+
+                try
+                {
+                    var receiveTasks = new Task<(int messagesReceived, long lastSequenceNumber)>[numPartitions];
+                    var sw = Stopwatch.StartNew();
+                    for (var i = 0; i < numPartitions; i++)
+                    {
+                        receiveTasks[i] = ReceiveAllMessages(clients[i % numClients], partitions[i]);
+                    }
+                    var results = await Task.WhenAll(receiveTasks);
+                    sw.Stop();
+
+                    var elapsed = sw.Elapsed.TotalSeconds;
+                    var messagesReceived = results.Select(r => r.messagesReceived).Sum();
+                    var messagesPerSecond = messagesReceived / elapsed;
+                    var megabytesPerSecond = (messagesPerSecond * _bytesPerMessage) / (1024 * 1024);
+
+                    Console.WriteLine($"Received {messagesReceived} messages of size {_bytesPerMessage} in {elapsed:N2}s " +
+                        $"({messagesPerSecond:N2} msg/s, {megabytesPerSecond:N2} MB/s)");
+                }
+                finally
+                {
+                    foreach (var consumer in clients)
+                    {
+                        await consumer.DisposeAsync();
+                    }
+                }
+            }
+            finally
+            {
+                foreach (var client in clients)
+                {
+                    await client.DisposeAsync();
+                }
+            }
+        }
+
+        private static async Task<(int messagesReceived, long lastSequenceNumber)> ReceiveAllMessages(EventHubConsumerClient consumer, PartitionProperties partition)
+        {
+            var messagesReceived = 0;
+            var lastSequenceNumber = (long)-1;
+            while (lastSequenceNumber < partition.LastEnqueuedSequenceNumber)
```

Comment:
This is an odd structure and I don't think it's going to do what you intend.    I think you'll want this logic inside your `foreach` block and use it to break out of the iterator when you meet this condition.   Otherwise, you're never going to exit.